### PR TITLE
Add some new functionality to obelisk-route to support simple paramet…

### PR DESCRIPTION
**Pull request by ryantrinkle**
…ers in the URL path

### CODE REVIEW

File: lib/route/src/Obelisk/Route.hs [hunk 1]
```diff
@@ -200,6 +205,48 @@ mapSome f (Some a) = Some $ f a
...
+--------------------------------------------------------------------------------
+-- Dealing with pairs (i.e. non-dependently-typed subroutes/paths)
+--------------------------------------------------------------------------------
+
+infixr 5 :.
+type (:.) = (,)
+
+{-# COMPLETE (:.) #-}
+pattern (:.) :: a -> b -> a :. b
+pattern a :. b = (a, b)
```
Comment (3noch) [inline lib/route/src/Obelisk/Route.hs]: Note that this does conflict with the same name in `postgresql-simple`.

File: lib/route/src/Obelisk/Route.hs [hunk 2]
```diff
@@ -200,6 +205,48 @@ mapSome f (Some a) = Some $ f a
 hoistR :: (forall x. f x -> g x) -> R f -> R g
 hoistR f (x :=> Identity y) = f x :/ y
 
+--------------------------------------------------------------------------------
+-- Dealing with pairs (i.e. non-dependently-typed subroutes/paths)
+--------------------------------------------------------------------------------
+
+infixr 5 :.
+type (:.) = (,)
```
Comment (3noch) [inline lib/route/src/Obelisk/Route.hs]: What value does this bring over just using tuples?

Comment (3noch) [reply lib/route/src/Obelisk/Route.hs]: It looks like it's only used in one place?

Comment (ryantrinkle) [reply lib/route/src/Obelisk/Route.hs]: When people use it right-associatively, it looks nicer than nested tuples.  Without something like this, people will use tuples with more entries than pairs, which can't be broken down very easily by other primitives.

Comment (3noch) [reply lib/route/src/Obelisk/Route.hs]: Ah makes sense thanks.

**Pull request merged**